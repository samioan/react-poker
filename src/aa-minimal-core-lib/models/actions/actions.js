import ActionCreator from "./ActionCreator";

const namespacedActionCreator = ActionCreator("@@core");

const effect = namespacedActionCreator("effect");

export { effect };
