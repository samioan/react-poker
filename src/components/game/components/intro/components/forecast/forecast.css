.container-f {
  display: flex;
  height: 10vh;
  flex-direction: row;
  justify-content: space-around;
}
.image {
  height: 4em;
  width: 4em;
  padding-left: 2em;
  padding-right: 2em;
  padding-top: 2em;
  padding-bottom: 2em;
}
.city-icon {
  content: url("cityIcon.png");
}
.temp-icon {
  content: url("tempIcon.png");
}
.weather-icon {
  content: url("weatherIcon.png");
}
.text-f {
  color: var(--buttonColor);
  padding-left: 0.3em;
  padding-right: 0.3em;
  padding-top: 0.3em;
  padding-bottom: 0.3em;
  text-align: center;
  font-size: 32px;
  text-decoration: none;
}
