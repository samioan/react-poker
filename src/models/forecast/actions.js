import ActionCreator from "aa-minimal-core-lib/models/actions/ActionCreator";

const namespacedActionCreator = ActionCreator("//FORECAST");

export const getForecast = namespacedActionCreator("GET_FORECAST");
