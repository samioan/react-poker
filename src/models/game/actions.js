const Action = (type) => {
  const actionCreator = (payload) => ({
    type,
    payload,
  });
  actionCreator.type = type;
  return actionCreator;
};

export const check = Action("CHECK");
export const playerChecked = Action("PLAYER_CHECKED");
export const playerWon = Action("PLAYER_WON");
export const playerLost = Action("PLAYER_LOST");
export const playerTied = Action("PLAYER_TIED");
export const startGame = Action("START_GAME");
export const gameStarted = Action("GAME_STARTED");
export const fold = Action("FOLD");
export const playerFolded = Action("PLAYER_FOLDED");
export const raise = Action("RAISE");
export const betRaised = Action("BET_RAISED");
export const replace = Action("REPLACE");
export const cardReplaced = Action("CARD_REPLACED");
export const nextPhase = Action("NEXT_PHASE");
export const phaseAdvanced = Action("PHASE_ADVANCED");
