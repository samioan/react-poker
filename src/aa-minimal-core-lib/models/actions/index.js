import ActionCreator from "./ActionCreator";
import { effect } from "./actions";

export { ActionCreator, effect };
