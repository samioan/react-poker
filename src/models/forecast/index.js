import { getForecast } from "./actions";

import { city, temperature, weather } from "./selectors";

export { getForecast, city, temperature, weather };
