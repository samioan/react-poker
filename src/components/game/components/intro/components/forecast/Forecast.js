import { Button } from "components/game/components/board/components";
import React, { useState } from "react";

import withForecastProps from "./withForecastProps";

//import { name, temp, description } from "models/forecast/selectors";

import "./forecast.css";

//const Forecast = ({ name, temp, description, onClickForecastHandler }) => (
//  <div>
//    <div>
//      <Button id={"Weather Forecast"} onClick={onClickForecastHandler} />
//    </div>
//    <div>
//      <p className="text">{name}</p>
//      <p className="text">{temp}</p>
//      <p className="text">{description}</p>
//    </div>
//  </div>
//);

const Conditions = (props) => {
  return (
    <div className="buttons-row">
      {props.responseObj.cod === 200 ? (
        <div className="buttons-row">
          <p className="text">
            Here in {props.responseObj.name} it is currently{" "}
            {Math.round(props.responseObj.main.temp)} degrees out with{" "}
            {props.responseObj.weather[0].description}.
          </p>
          {Math.round(props.responseObj.main.temp) <= 16 ? (
            <p className="text">
              Grab some hot chocolate and play Poker because it's cold outside!
            </p>
          ) : null}
        </div>
      ) : null}
    </div>
  );
};

const Forecast = () => {
  let [responseObj, setResponseObj] = useState({});

  function getForecast() {
    fetch(
      "https://community-open-weather-map.p.rapidapi.com/weather?q=Athens%2Cgr&units=metric",
      {
        method: "GET",
        headers: {
          "x-rapidapi-key":
            "7f268e1e4amsh5918349f7b2b61bp1e32b3jsnf97c9be51f24",
          "x-rapidapi-host": "community-open-weather-map.p.rapidapi.com",
        },
      }
    )
      .then((response) => response.json())
      .then((response) => {
        setResponseObj(response);
      })
      .catch((err) => {
        console.error(err);
      });
  }
  return (
    <div className="container-f">
      <div className="buttons-row">
        <Button id={"Today's Weather Forecast"} onClick={getForecast} />
      </div>
      <div className="buttons-row">
        <Conditions responseObj={responseObj} />
      </div>
    </div>
  );
};

export { Forecast };
export default withForecastProps(Forecast);
