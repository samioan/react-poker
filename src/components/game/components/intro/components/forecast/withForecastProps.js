import React from "react";
import { connect } from "react-redux";
import { compose } from "redux";

import { getForecast } from "models/forecast/actions";
import { name, temp, description } from "models/forecast/selectors";

import "./forecast.css";

const withForecastProps = (Component) => (props) => {
  const { name, temp, description } = props;

  const newProps = {
    ...props,
    showMessage:
      temp < 16 ? "It's gonna be a cold day!" : "A nice and warm day!",
  };

  return <Component {...newProps} />;
};

const mapStateToProps = (state) => ({
  name: name(state),
  temp: temp(state),
  description: description(state),
});

const mapDispatchToProps = (dispatch) => ({
  onClickForecastHandler: () => dispatch(getForecast()),
});

export { withForecastProps };
export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withForecastProps
);
