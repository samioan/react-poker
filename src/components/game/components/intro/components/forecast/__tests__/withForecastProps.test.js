import { withForecastProps } from "../withForecastProps";
import * as forecastActions from "models/forecast/actions";
import * as react from "react";

const DummyComponent = (props) => <div>{props}</div>;

describe("withForecastProps HOC", () => {
  let useEffectSpy;

  beforeEach(() => {
    useEffectSpy = jest
      .spyOn(react, "useEffect")
      .mockImplementation((f) => f());
  });

  afterEach(() => {
    useEffectSpy.mockRestore();
  });

  test("renders correctly", () => {
    // GIVEN
    const mockProps = {
      getForecastOnLoad: jest.fn(() => {}),
      temp: 15,
    };
    const EnchancedComponent = withForecastProps(DummyComponent);

    // WHEN
    const tree = shallowSnapshot(<EnchancedComponent {...mockProps} />);

    // THEN
    expect(tree).toMatchSnapshot();
    expect(mockProps.getForecastOnLoad).toHaveBeenCalled();
  });
});
