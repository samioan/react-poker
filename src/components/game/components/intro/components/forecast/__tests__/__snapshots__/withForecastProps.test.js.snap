// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withForecastProps HOC renders correctly 1`] = `
<DummyComponent
  getForecastOnLoad={[MockFunction]}
  showMessage="A cold day!"
  temp={15}
/>
`;
